pkg/app/piped/eventwatcher/eventwatcher.go

```diff
@@ -428,10 +431,13 @@ func (w *watcher) execute(ctx context.Context, repo git.Repo, repoID string, eve
 	}
 
 	retry := backoff.NewRetry(retryPushNum, backoff.NewConstant(retryPushInterval))
-	_, err = retry.Do(ctx, func() (interface{}, error) {
-		err := tmpRepo.Push(ctx, tmpRepo.GetClonedBranch())
-		return nil, err
-	})
+	for branch := range branchNames {
+		_, err = retry.Do(ctx, func() (interface{}, error) {
+			err := tmpRepo.Push(ctx, branch)
+			return nil, err
+		})
+	}
```

Comment:
If one of the pushes to remote branch fails, the entry handledEvents will be marked as FAILURE due to these push branch action use the same error 🤔

Reply:
Note sure is that the best solution, but I think we have to make this "post action" for each `tmpRepo.Push` call. So we may need a map of kind map[string]HandledEvents, in which key is the branchName and the value will be handledEvents. Then for each push to remote branch run, we will update the handledEvents status to success or failed and so on. wdyt? 🤔

Reply:
Thank you!!!
I think so, too.
I used `map[string]HandledEvents` and added "post action" for each `tmpRepo.Push` call.
[use branchHandledEvents map](https://github.com/pipe-cd/pipecd/pull/4395/commits/db2f974a02fd7b69d2cb8d9a23e7c04ff3a5b782)